scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -1729,12 +1729,18 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
           Type.Macro(macroSplice())
         case Ident("?") if dialect.allowQuestionMarkPlaceholder =>
           next(); atPos(in.prevTokenPos, auto)(Type.Placeholder(typeBounds()))
+        case ident: Ident
+            if (ident.value == "+" || ident.value == "-") && ahead(
+              token.is[Underscore]
+            ) && dialect.allowPlusMinusBeforeUndescore =>
+          autoPos {
+            accept[Ident]
+            accept[Underscore]
+            Type.Name(s"${ident.value}_")
+          }
         case Literal() =>
           if (dialect.allowLiteralTypes) literal()
           else syntaxError(s"$dialect doesn't support literal types", at = path())
-        case Ident("-") if ahead { token.is[NumericLiteral] } && dialect.allowLiteralTypes =>
```

Comment:
Why remove this one? It's unrelated to the new syntax

Reply:
oops :facepalm:

Reply:
do we need to add a test for this, so that no accidental regressions?

Reply:
no, actually I was distracted by GitHub actions failure so I didn't look on the diff.